File: Assignment_NET104/wwwroot/js/fooditem.js
$(document).on('click', '[id^="my-btn-"]', function () {
    let itemId = $(this).attr('id').split('-')[2];

    let item = {
        FoodItemId: parseInt($('#itemId-' + itemId).val()),
        Name: $('#itemName-' + itemId).val(),
        ImagePath: $('#itemImage-' + itemId).val(),
        Price: parseFloat($('#itemPrice-' + itemId).val()),
        Category: Categories[$('#itemType-' + itemId).val()],
        Status: 1
    }

    console.log(item);

    $.ajax({
        type: 'POST',
        url: '/Cart/AddToCart',
        contentType: 'application/json',
        data: JSON.stringify(item),
        dataType: 'json',
        success: function (data) {
            alert(data.status);
        },
        error: function (xhr, status, error) {
            console.log("AJAX Error: ", status, error);
            console.log("Response Text: ", xhr.responseText);
        }
    });
});

const Categories = {
    MonAn: 1,
    Combo: 2,
    Nuoc: 3,
    Khac: 4
};

File: Assignment_NET104/Models/FoodItem.cs
namespace Assignment_NET104.Models
{
    public enum Categories
    {
        MonAn = 1,
        Combo = 2,
        Nuoc = 3,
        Khac = 4
    }

    public enum Status
    {
        Ready = 1,
        OutOfStock = 2
    }

    public class FoodItem
    {
        public int FoodItemId { get; set; }
        public string Name { get; set; } = string.Empty;
        public Categories Category { get; set; }
        public string? ImagePath { get; set; }
        public IFormFile? ImageFile { get; set; }
        public decimal Price { get; set; }
        public Status Status { get; set; }

        // Relationship
        public ICollection<CartItem> CartItems { get; set; } = new List<CartItem>();
        public ICollection<OrderItem> OrderDetails { get; set; } =  new List<OrderItem>();
    }
}
